cpp/src/arrow/python/pandas_to_arrow.cc

```diff
@@ -944,10 +944,6 @@ inline Status PandasConverter::ConvertTypedLists(const std::shared_ptr<DataType>
     return Status::NotImplemented("mask not supported in object conversions yet");
   }
 
-  if (is_strided()) {
```

Comment:
It looks like not. I'll add one, since we should be using https://github.com/apache/arrow/blob/f7ab7270bb07466dabf84c015a6db2a192eb3dad/cpp/src/arrow/python/numpy-internal.h#L34 for this iteration rather than going directly to `PyArray_DATA`